exchanges/kraken/kraken.go

```diff
@@ -291,74 +256,36 @@ func (k *Kraken) GetDepth(ctx context.Context, symbol currency.Pair) (*Orderbook
 	values := url.Values{}
 	values.Set("pair", symbolValue)
 	path := fmt.Sprintf("/%s/public/%s?%s", krakenAPIVersion, krakenDepth, values.Encode())
-	var result interface{}
-	err = k.SendHTTPRequest(ctx, exchange.RestSpot, path, &result)
-	if err != nil {
-		return nil, err
-	}
 
-	if result == nil {
-		return nil, fmt.Errorf("%s GetDepth result is nil", k.Name)
+	type orderbookStructure struct {
+		Bids [][]types.Number `json:"bids"`
+		Asks [][]types.Number `json:"asks"`
```

Comment:
```suggestion
		Bids [][3]types.Number `json:"bids"`
		Asks [][3]types.Number `json:"asks"`
```
Not huge, but allows your assumptions on struct length to work since dropping len check

Reply:
Done! Big fan of preallocating where we can